From 55468a8a53dea55815dac5262661d038ace9a134 Mon Sep 17 00:00:00 2001
From: Boyuan Yang <byang@debian.org>
Date: Fri, 23 Jan 2026 09:27:41 -0500
Subject: [PATCH] Use pre-python3.14 fork method for multiprocessing.Process

This keeps the old behavior of multiprocessing module and avoids
testsuite failures when using Python 3.14 or later Python versions.

This patch closes #88.
---
 timeout_decorator/timeout_decorator.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/timeout_decorator/timeout_decorator.py b/timeout_decorator/timeout_decorator.py
index 42b6686..2263170 100644
--- a/timeout_decorator/timeout_decorator.py
+++ b/timeout_decorator/timeout_decorator.py
@@ -140,9 +140,10 @@ def __call__(self, *args, **kwargs):
         self.__limit = kwargs.pop('timeout', self.__limit)
         self.__queue = multiprocessing.Queue(1)
         args = (self.__queue, self.__function) + args
-        self.__process = multiprocessing.Process(target=_target,
-                                                 args=args,
-                                                 kwargs=kwargs)
+        ctx = multiprocessing.get_context('fork')
+        self.__process = ctx.Process(target=_target,
+                                     args=args,
+                                     kwargs=kwargs)
         self.__process.daemon = True
         self.__process.start()
         if self.__limit is not None:
